packages/client/src/apis/rewards/RewardsExecuteAPI.ts

```diff
@@ -93,33 +137,85 @@ export class RewardsExecuteAPI extends BaseVertexAPI {
   }
 
   /**
-   * Claim unlocked tokens that were previously unstaked
+   * Claim unlocked V2 tokens that were previously unstaked
    */
-  async withdrawUnstakedTokens() {
-    return this.context.contracts.vrtxStaking.write.claimVrtx();
+  async withdrawUnstakedV2Tokens() {
+    return this.context.contracts.vrtxStakingV2.write.claimWithdraw();
   }
 
   /**
-   * Claim unlocked V2 tokens that were previously unstaked
+   *
+   *  Staking Satellite
+   *
    */
-  async withdrawUnstakedV2Tokens() {
-    return this.context.contracts.vrtxStakingV2.write.claimWithdraw();
+
+  /**
+   * Stake V2 VRTX tokens on non-canonical chains
+   */
+  async stakeV2Satellite(params: CcipStakingParams) {
+    const address = this.getWalletClientAddress();
+
+    return this.context.contracts.vrtxStakingV2Satellite.write.stakeAs(
+      [address, toBigInt(params.amount)],
+      { value: toBigInt(params.ccipFee) },
+    );
   }
 
   /**
-   * Claim staking rewards (in USDC)
+   * Unstake V2 VRTX tokens on non-canonical chains, unstake tokens instantly with a penalty that is redistributed
+   * to the staking contract
    */
-  async claimStakingRewards() {
-    return this.context.contracts.vrtxStaking.write.claimUsdc();
+  async unstakeV2Satellite(params: CcipParams) {
+    return this.context.contracts.vrtxStakingV2Satellite.write.withdraw({
+      value: toBigInt(params.ccipFee),
+    });
   }
 
   /**
-   * Claim staking rewards (in USDC), swap for VRTX, and stake the VRTX
+   * Unstake V2 VRTX tokens on non-canonical chains, unstaked tokens that are unlocked will need to be claimed
+   * after 21 day locking period
    */
-  async claimAndStakeStakingRewards() {
-    return this.context.contracts.vrtxStaking.write.claimUsdcAndStake();
+  async unstakeV2SatelliteSlow(params: CcipParams) {
+    return this.context.contracts.vrtxStakingV2Satellite.write.withdrawSlow({
+      value: toBigInt(params.ccipFee),
+    });
+  }
+
+  /**
+   * Claim unlocked V2 tokens that were previously unstaked on non-canonical chains
+   */
+  async withdrawUnstakedV2SatelliteTokens(params: CcipParams) {
+    return this.context.contracts.vrtxStakingV2Satellite.write.claimWithdraw({
+      value: toBigInt(params.ccipFee),
+    });
   }
 
+  /**
+   * Claim staking rewards (in USDC) on non-canonical chains
```

Comment:
Not staking rewards - trading rewards (should probably be grouped with the `claimLiquidTokens` fn)